### ISSUE TITLE
Make this package PEP-561 compatible by adding a py.typed marker

### ISSUE THREAD

**Issue by Priyansh121096** [trinodb/trino-python-client#482]
### Describe the feature

I see that static type checking was discussed in #69 and mypy was enabled in #91. Do people think it's a good idea to add a [py.typed marker](https://peps.python.org/pep-0561/#specification) so that packages depending on `trino` can type check themselves against it? 

Right now, there's no way to force mypy to type check against this site-packages which lack this marker ([ref](https://github.com/python/mypy/issues/8545)).

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by hashhar** [comment]
Good idea, if you are still willing to send a PR we'd appreciate it. Sorry it took so long to reply.
